src/openrct2/actions/MazeSetTrackAction.hpp

```diff
@@ -259,12 +259,12 @@ struct MazeSetTrackAction : public GameActionBase<GAME_COMMAND_SET_MAZE_TRACK, G
             {
                 uint8_t segmentOffset = MazeGetSegmentBit(_x, _y);
 
-                tileElement->properties.track.maze_entry &= ~(1 << segmentOffset);
+                tileElement->AsTrack()->MazeEntrySubtract(1 << segmentOffset);
```

Comment:
I think the bitshift is something implementation-specific, right? I'm inclined to say the track function should handle it.